crates/holochain/src/conductor/conductor.rs

```diff
@@ -185,20 +175,34 @@ pub enum CellStatus {
     Joining,
 }
 
+/// The reason why a cell is waiting to join the network.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum PendingJoinReason {
+    /// The initial state, no attempt has been made to join the network yet.
+    Initial,
+
+    /// The join failed with an error that is safe to retry, such as not
+    /// being connected to the internet.
+    Retry(String),
+
+    /// The network join failed and will not be retried. This will impact
+    /// the status of the associated
+    /// app and require manual intervention from the user.
+    Failed(String),
+
+    /// The join attempt has timed out.
+    TimedOut,
+}
+
 /// A [`Cell`] tracked by a Conductor, along with its [`CellStatus`]
-#[derive(Debug, Clone, Serialize)]
-#[serde(into = "CellStatus")]
```

Comment:
Should the serde customisation not stay for serialisation or was it always just cosmetic?

Reply:
I had added the serialization recently for DumpConductorState, mainly to be able to dump the CellStatus, but now that that's gone (including from the dump), there's no reason to serialize this anymore.